docs/source/preparing_datasets/basic_dataset_conversion.md

```diff
@@ -0,0 +1,330 @@
+
+1. An `out` parameter is an output directory to save shard files. The `out` directory can be specified in three ways:
+ * **Local path**: Shard files are stored locally.
+ * **Remote path**: A local temporary directory is created to cache the shard files, and when shard creation is complete, they are uploaded to the remote location.
+ * **`(local_dir, remote_dir)` tuple**: Shard files are saved in the specified `local_dir` and uploaded to `remote_dir`.
+
+<!--pytest.mark.skip-->
+```python
+out = '/local/data'
+out = 's3://bucket/data' # Will create a temporary local dir
+out = ('/local/data', 'oci://bucket/data')
+```
+
+2. The optional `keep_local` parameter controls if you would like to keep the shard files locally after they have been uploaded to a remote cloud location. To save local disk space, this defaults to `False`.
+
+3. A `column` parameter is a `dict` mapping a feature name or label name with a streaming supported encoding type. `MDSWriter` encodes your data to bytes, and at training time, data gets decoded back automatically to its original form. The `index.json` file stores `column` metadata for decoding. Supported encoding formats are:
+
+| Category           | Name          | Class        | Notes                    |
+|--------------------|---------------|--------------|--------------------------|
+| Encoding           | 'bytes'       | `Bytes`      | no-op encoding           |
+| Encoding           | 'str'         | `Str`        | stores in UTF-8          |
+| Encoding           | 'int'         | `Int`        | uses `numpy.int64`       |
+| Numpy Array        | 'ndarray'     | `NDArray`    | uses `numpy.ndarray`     |
+| Numpy Unsigned Int | 'uint8'       | `UInt8`      | uses `numpy.uint8`       |
+| Numpy Unsigned Int | 'uint16'      | `UInt16`     | uses `numpy.uint16`      |
+| Numpy Unsigned Int | 'uint32'      | `Uint32`     | uses `numpy.uint32`      |
+| Numpy Unsigned Int | 'uint64'      | `Uint64`     | uses  `numpy.uint64`     |
+| Numpy Signed Int   | 'int8'        | `Int8`       | uses  `numpy.int8`       |
+| Numpy Signed Int   | 'int16'       | `Int16`      | uses  `numpy.int16`      |
+| Numpy Signed Int   | 'int32'       | `Int32`      | uses  `numpy.int32`      |
+| Numpy Signed Int   | 'int64'       | `Int64`      | uses  `numpy.int64`      |
+| Numpy Float        | 'float16'     | `Float16`    | uses  `numpy.float16`    |
+| Numpy Float        | 'float32'     | `Float32`    | uses  `numpy.float32`    |
+| Numpy Float        | 'float64'     | `Float64`    | uses  `numpy.float64`    |
+| Numerical String   | 'str_int'     | `StrInt`     | stores in UTF-8          |
+| Numerical String   | 'str_float'   | `StrFloat`   | stores in UTF-8          |
+| Numerical String   | 'str_decimal' | `StrDecimal` | stores in UTF-8          |
+| Image              | 'pil'         | `PIL`        | raw PIL image            |
```

Comment:
when does one prefer `pil` vs `png` vs `jpeg`? what does raw mean?

Reply:
Whether the user prefers to use it or not is up to them, raw meaning this class. Will add [link](https://pillow.readthedocs.io/en/stable/reference/Image.html)

Reply:
Like it's just whether or not a user has png, jpeg, or uses the PIL image class directly